### ISSUE TITLE
Conflicts between open-solution-salt-identification and ipython

### ISSUE THREAD

**Issue by NeolithEra** [neptune-ml/open-solution-salt-identification#95]
Hi, users are unable to run open-solution-salt-identification due to dependency conflict with _**ipython**_ package. 
As shown in the following full dependency graph of  open-solution-salt-identification, open-solution-salt-identification requires _**ipython==6.3.1**_, while **steppy==0.1.6** requires _**ipython >= 6.4.0**.
According to pip’s “first found wins” installation strategy, _**ipython 6.3.1**_ is the actually installed version. However, _**ipython 6.3.1**_ does not satisfy _**>=6.4.0**_.

### Dependency tree------
```
open-solution-salt-identification-master<version range:>
| +- attrdict<version range:==2.0.0>
| | +- six<version range:>
| +- cffi<version range:==1.11.5>
| +- click<version range:==6.7>
| +- cython<version range:==0.28.2>
| +- imageio<version range:==2.2.0>
| | +- numpy<version range:>
| | +- pillow<version range:>
| +- imgaug<version range:==0.2.5>
| | +- numpy<version range:>=1.7.0>
| | +- scikit-image<version range:>=0.11.0>
| | +- scipy<version range:>
| | +- six<version range:>
| +- ipython<version range:==6.3.1>
| +- kaggle<version range:==1.3.11.1>
| | +- certifi<version range:>
| | +- python-dateutil<version range:>
| | +- requests<version range:>
| | +- six<version range:>=1.10>
| | +- tqdm<version range:>
| | +- urllib3<version range:>=1.15,<1.23.0>
| +- neptune-cli<version range:>
| | +- argcomplete<version range:>=1.9.3>
| | +- enum34<version range:>=1.1.6>
| | +- flask<version range:==0.12.0>
| | +- future<version range:>=0.16.0>
| | +- gitpython<version range:>=2.0.8>
| | +- humanize<version range:>=0.5.1>
| | +- kitchen<version range:>=1.2.4>
| | +- more-itertools<version range:>=4.1.0>
| | +- nvidia-ml-py3<version range:>=7.352.0>
| | +- pathlib2<version range:==2.3.0>
| | +- pillow<version range:>=1.7.6>
| | +- psutil<version range:>=5.4.3>
| | +- pyjwt<version range:>=1.5.2>
| | +- pykwalify<version range:<1.6.0,>=1.5.2>
| | | +- docopt<version range:>=0.6.2>
| | | +- python-dateutil<version range:>=2.4.2>
| | | +- pyyaml<version range:>=3.11>
| | +- raven<version range:>=6.1.0>
| | +- requests<version range:>=2.11.1>
| | +- requests-oauthlib<version range:>=0.8.0>
| | +- terminaltables<version range:>=2.1.0,<3.0.0>
| | +- tqdm<version range:>=4.11.2>
| | +- voluptuous<version range:>=0.9.3>
| | +- websocket-client<version range:>=0.35.0>
| +- numpy<version range:==1.14.0>
| +- opencv-python<version range:==3.4.0.12>
| +- pandas<version range:==0.22.0>
| +- pillow<version range:==5.1.0>
| +- pretrainedmodels<version range:==0.7.0>
| | +- munch<version range:>
| | | +- six<version range:>
| | +- torch<version range:>
| | +- torchvision<version range:>
| +- pycocotools<version range:==2.0.0>
| +- pydot-ng<version range:==1.0.0>
| | +- pyparsing<version range:>=2.0.1>
| +- pyyaml<version range:>=4.2b1>
| +- scikit-image<version range:==0.13.1>
| +- scikit-learn<version range:==0.19.2>
| +- scipy<version range:==1.0.0>
| +- steppy<version range:==0.1.6>
| | +- ipython<version range:>=6.4.0>
| | +- numpy<version range:>=1.14.0>
| | +- pydot-ng<version range:>=1.0.0>
| | | +- pyparsing<version range:>=2.0.1>
| | +- pytest<version range:>=3.6.0>
| | +- scikit-learn<version range:>=0.19.0>
| | +- scipy<version range:>=1.0.0>
| | +- setuptools<version range:>=39.2.0>
| | +- typing<version range:>=3.6.4>
| +- steppy-toolkit<version range:==0.1.5>
| | +- attrdict<version range:>=2.0.0>
| | | +- six<version range:>
| | +- neptune-cli<version range:>=2.8.5>
| | | +- argcomplete<version range:>=1.9.3>
| | | +- enum34<version range:>=1.1.6>
| | | +- flask<version range:==0.12.0>
| | | +- future<version range:>=0.16.0>
| | | +- gitpython<version range:>=2.0.8>
| | | +- humanize<version range:>=0.5.1>
| | | +- kitchen<version range:>=1.2.4>
| | | +- more-itertools<version range:>=4.1.0>
| | | +- nvidia-ml-py3<version range:>=7.352.0>
| | | +- pathlib2<version range:==2.3.0>
| | | +- pillow<version range:>=1.7.6>
| | | +- psutil<version range:>=5.4.3>
| | | +- pyjwt<version range:>=1.5.2>
| | | +- pykwalify<version range:<1.6.0,>=1.5.2>
| | | | +- docopt<version range:>=0.6.2>
| | | | +- python-dateutil<version range:>=2.4.2>
| | | | +- pyyaml<version range:>=3.11>
| | | +- raven<version range:>=6.1.0>
| | | +- requests<version range:>=2.11.1>
| | | +- requests-oauthlib<version range:>=0.8.0>
| | | +- terminaltables<version range:>=2.1.0,<3.0.0>
| | | +- tqdm<version range:>=4.11.2>
| | | +- voluptuous<version range:>=0.9.3>
| | | +- websocket-client<version range:>=0.35.0>
| | +- numpy<version range:>=1.14.3>
| | +- pandas<version range:>=0.23.0>
| | +- pytest<version range:>=3.6.0>
| | +- setuptools<version range:>=39.2.0>
| | +- steppy<version range:>=0.1.4>
| | | +- ipython<version range:>=6.4.0>
| | | +- numpy<version range:>=1.14.0>
| | | +- pydot-ng<version range:>=1.0.0>
| | | | +- pyparsing<version range:>=2.0.1>
| | | +- pytest<version range:>=3.6.0>
| | | +- scikit-learn<version range:>=0.19.0>
| | | +- scipy<version range:>=1.0.0>
| | | +- setuptools<version range:>=39.2.0>
| | | +- typing<version range:>=3.6.4>
| +- torch<version range:==0.3.1>
| +- torchvision<version range:==0.2.0>
| +- tqdm<version range:==4.23.0>

```

### Solution
Could you please loose your version of ipython (change ipython==6.3.1 to >=6.3.1)?
Thanks for your help.
Best,
Neolith

**Comment by NeolithEra** [comment]
Could you help me review this issue?

**Comment by jakubczakon** [comment]
Thanks @NeolithEra thanks for bringing that up, 
I will take a look and update this repo. 
There are is also neptune-cli dependency which I have to update to neptune-client.

**Comment by jakubczakon** [comment]
@NeolithEra I have updated instructions and switched from pip to conda to make things simpler.
You should be able to start working with no problems.

Please let me know how it went.

**Comment by NeolithEra** [comment]
Commit [d235f939c1b4034e91549187b5b22837f5e2d276](https://github.com/neptune-ml/open-solution-salt-identification/commit/d235f939c1b4034e91549187b5b22837f5e2d276) fixes issue #95 and #96.

**Comment by NeolithEra** [comment]
@jakubczakon Thank you very much. These two issues can be closed.
